pkg/validation/app.go

```diff
@@ -367,7 +367,9 @@ func (v *Validator) validateMetadataConstraints(ctx context.Context, cr v1alpha1
 
 	for _, app := range apps {
 		if app.Spec.Name == cr.Spec.Name {
-			if entry.Spec.Restrictions.ClusterSingleton {
+			if entry.Spec.Restrictions.ClusterSingleton &&
```

Comment:
Can you move that out in a function? something like isValidClusterSingleton and check that if in org-namespace, we check the kubeconfgi otherwise we do not?

Reply:
But we can also merge now and let honeybadger improve :D

Reply:
The only concern, I have is the `org-*`

Reply:
I believe, I saw some thread mentioning that the org prefix will be removed

Reply:
That;s also a way yes :)

Reply:
Never mind, I just used `key.IsInOrgNamespace(cr)`